src/Symfony/Bundle/TwigBundle/DependencyInjection/Configuration.php

```diff
@@ -127,7 +127,13 @@ private function addTwigOptions(ArrayNodeDefinition $rootNode)
                 ->scalarNode('cache')->defaultValue('%kernel.cache_dir%/twig')->end()
                 ->scalarNode('charset')->defaultValue('%kernel.charset%')->end()
                 ->booleanNode('debug')->defaultValue('%kernel.debug%')->end()
-                ->booleanNode('strict_variables')->end()
+                ->booleanNode('strict_variables')
+                    ->defaultValue(function () {
+                        @trigger_error('No setting "strict_variables" to false is deprecated as of 4.1, it will be the value of the "kernel.debug" parameter by default in 5.0.', E_USER_DEPRECATED);
```

Comment:
@nicolas-grekas done, but I'm not completely sure if the numbers are correct: `4.1` ... `5.0` ?

Reply:
Btw, is it the right way to do it? How I can avoid the deprecation notices in tests?

Reply:
You would need to configure a value explicitly instead of relying on the default value.

Reply:
Thanks @xabbuh, I've made the changes, but in some places it seems out of place, so we need remember remove it relying on the default value again in 5.0. The question is: how to remember that? what is the common practice in these case? adding a direct comment to the code?

Reply:
in the past, we often added inline comments like "to be removed when ..." (at least, I used this phrase a lot to search for code that could be removed when we started the development on 4.0)

Reply:
Inline comments added, thanks.

Reply:
We are now using `is deprecated as of Symfony 4.1`